Snakefile

```diff
@@ -55,47 +64,60 @@ rule all:
         ],
 
 
-def devtools_install():
-    if config["ploidetect_local_clone"] and config["ploidetect_local_clone"] != "None":
-        install_path = config["ploidetect_local_clone"].format(**config)
-        devtools_cmd = f"devtools::install_local('{install_path}', force = TRUE)"
-    else:
-        ver = config["ploidetect_ver"]
-        devtools_cmd = f"devtools::install_github('lculibrk/Ploidetect', ref = '{ver}')"
-    return f'"{devtools_cmd}"'
-
 rule download_cytobands:
     """Downloads cytoband data for plotting & (todo) hgver-specific centromere filtering"""
     input:
         HTTP.remote(
-            expand("http://hgdownload.cse.ucsc.edu/goldenpath/{hgver}/database/cytoBand.txt.gz", hgver = config["genome_name"])
-        )
+            expand(
+                "http://hgdownload.cse.ucsc.edu/goldenpath/{hgver}/database/cytoBand.txt.gz",
+                hgver=config["genome_name"],
+            )
+        ),
     output:
-        expand("resources/{hgver}/cytobands.txt", hgver = config["genome_name"])
+        expand("resources/{hgver}/cytobands.txt", hgver=config["genome_name"]),
     resources:
         cpus=1,
         mem_mb=MEM_PER_CPU,
+    conda:
+        "conda_configs/sequence_processing.yaml"
     shell:
         "gunzip -c {input} > {output}"
 
+
+ploidetect_install_cmd = (
+    "devtools::install_github('lculibrk/Ploidetect', ref = '{}')".format(
+        config["ploidetect_ver"]
+    )
+)
+if config["ploidetect_local_clone"] and config["ploidetect_local_clone"] != "None":
+    install_path = config["ploidetect_local_clone"].format(**config)
+    ploidetect_install_cmd = f"devtools::install_local('{install_path}', force = TRUE)"
+ploidetect_install_cmd = f'"{ploidetect_install_cmd}"'
+
+
 rule ploidetect_install:
     """Install Ploidetect R script into environment"""
     output:
         expand(
-            "{install_dir}/conda_configs/ploidetect_installed.txt",
+            "{install_dir}/conda_configs/ploidetect_installed_{ploidetect_ver}.txt",
             install_dir=workflow.basedir,
+            ploidetect_ver=config["ploidetect_ver"],
         ),
     resources:
         cpus=1,
         mem_mb=MEM_PER_CPU,
     conda:
         "conda_configs/r.yaml"
     params:
-        devtools_install(),
+        ploidetect_install_cmd,
+    log:
+        f"{workflow.basedir}/conda_configs/ploidetect_installed.log",
     shell:
+        "date >> {log}; echo {params} >> {log}; "
         "export LC_ALL=en_US.UTF-8; "
-        " Rscript -e {params} "
-        " && echo {params} > {output} && date >> {output}"
+        " Rscript -e {params} >> {log} 2>> {log}"
```

Comment:
I'm not actually that familiar with conda... does this rule depend on the the use having R script installed?

Reply:
The Ploidetect R package isn't on Bioconductor or CRAN, so it needs to be installed from github. There's no way to do this automatically in a conda spec file, so if the user is running the pipeline with conda, this rule installs the R package into the conda environment. The conda env has R/Rscript already installed, this rule just adds the Ploidetect package.